### PR TITLE
Add token Management controller

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: src/tokenManagement/tokenManagementTransactionsFactory.ts [hunk 1]
```diff
@@ -440,44 +362,44 @@ export class TokenManagementTransactionsFactory {
...
+            sender: sender,
+            receiver: sender,
             dataParts: dataParts,
             gasLimit: this.config.gasLimitEsdtNftCreate + storageGasLimit,
             addDataMovementGas: true,
         }).build();
     }
 
-    createTransactionForPausing(options: { sender: IAddress; tokenIdentifier: string }): Transaction {
+    createTransactionForPausing(sender: IAddress, options: resources.ManagementInput): Transaction {
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsFactory.ts:373]: `resources.ManagementInput` also has a mandatory `user: IAddress` field that this method doesn't need.

File: src/tokenManagement/tokenManagementTransactionsFactory.ts [hunk 2]
```diff
@@ -440,44 +362,44 @@ export class TokenManagementTransactionsFactory {
...
+            sender: sender,
+            receiver: sender,
             dataParts: dataParts,
             gasLimit: this.config.gasLimitPausing,
             addDataMovementGas: true,
         }).build();
     }
 
-    createTransactionForUnpausing(options: { sender: IAddress; tokenIdentifier: string }): Transaction {
+    createTransactionForUnpausing(sender: IAddress, options: resources.ManagementInput): Transaction {
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsFactory.ts:386]: the above comment applies here, as well

File: src/tokenManagement/tokenManagementTransactionsFactory.ts [hunk 3]
```diff
@@ -440,44 +362,44 @@ export class TokenManagementTransactionsFactory {
...
+            sender: sender,
+            receiver: sender,
             dataParts: dataParts,
             gasLimit: this.config.gasLimitPausing,
             addDataMovementGas: true,
         }).build();
     }
 
-    createTransactionForFreezing(options: { sender: IAddress; user: IAddress; tokenIdentifier: string }): Transaction {
+    createTransactionForFreezing(sender: IAddress, options: resources.ManagementInput): Transaction {
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsFactory.ts:402]: indeed, it is right here.

File: src/tokenManagement/tokenManagementTransactionsFactory.ts [hunk 4]
```diff
@@ -705,19 +590,15 @@ export class TokenManagementTransactionsFactory {
...
             addDataMovementGas: true,
         }).build();
     }
 
-    createTransactionForModifyingCreator(options: {
-        sender: IAddress;
-        tokenIdentifier: string;
-        tokenNonce: bigint;
-    }): Transaction {
+    createTransactionForModifyingCreator(sender: IAddress, options: resources.BaseInput): Transaction {
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsFactory.ts:601]: Not very explicit naming. Could've extended  `BaseInput` to have another name.

File: src/tokenManagement/tokenManagementTransactionsFactory.ts [hunk 5]
```diff
@@ -761,24 +633,15 @@ export class TokenManagementTransactionsFactory {
...
-        sender: IAddress;
-        tokenIdentifier: string;
-        tokenNonce: bigint;
-        newTokenName: string;
-        newRoyalties: bigint;
-        newHash: string;
-        newAttributes: Uint8Array;
-        newUris: string[];
-    }): Transaction {
+    createTransactionForMetadataRecreate(sender: IAddress, options: resources.UpdateMetadataInput): Transaction {
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsFactory.ts:644]: here the input name does not really match the method name

File: src/tokenManagement/tokenManagementTransactionsFactory.ts [hunk 6]
```diff
@@ -794,52 +657,53 @@ export class TokenManagementTransactionsFactory {
...
             dataParts: dataParts,
             gasLimit: this.config.gasLimitNftMetadataRecreate,
             addDataMovementGas: true,
         }).build();
     }
 
-    createTransactionForChangingTokenToDynamic(options: { sender: IAddress; tokenIdentifier: string }): Transaction {
+    createTransactionForChangingTokenToDynamic(
+        sender: IAddress,
+        options: resources.BurnRoleGloballyInput,
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsFactory.ts:670]: the input name does not match the method name

File: src/tokenManagement/tokenManagementTransactionsFactory.ts [hunk 7]
```diff
@@ -794,52 +657,53 @@ export class TokenManagementTransactionsFactory {
...
+            sender: sender,
             receiver: this.esdtContractAddress,
             dataParts: dataParts,
             gasLimit: this.config.gasLimitNftChangeToDynamic,
             addDataMovementGas: true,
         }).build();
     }
 
-    createTransactionForUpdatingTokenId(options: { sender: IAddress; tokenIdentifier: string }): Transaction {
+    createTransactionForUpdatingTokenId(sender: IAddress, options: resources.BurnRoleGloballyInput): Transaction {
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsFactory.ts:687]: the input name does not match the method name

File: src/tokenManagement/tokenManagementController.ts [hunk 8]
```diff
@@ -0,0 +1,495 @@
+import { IAccount } from "../accounts/interfaces";
+import { TransactionOnNetwork } from "../networkProviders";
+import { INetworkProvider } from "../networkProviders/interface";
+import { IESDTIssueOutcome } from "../tokenOperations";
```
Comment (popenta) [inline src/tokenManagement/tokenManagementController.ts:4]: the `tokenOperations` package will be removed and we should not import from there.

File: src/tokenManagement/tokenManagementController.ts [hunk 9]
```diff
@@ -0,0 +1,495 @@
...
+    }
+
+    parseAddQuantity(transactionOnNetwork: TransactionOnNetwork): { tokenIdentifier: string }[] {
+        return this.parser.parseAddQuantity(transactionOnNetwork);
+    }
+
+    async createTransactionForBurningQuantity(
+        sender: IAccount,
+        nonce: bigint,
+        options: resources.UpdateQuantityInput,
```
Comment (popenta) [inline src/tokenManagement/tokenManagementController.ts:477]: The input should be updated here, same as for the transactions factory.

File: src/tokenManagement/tokenManagementTransactionsFactory.ts (hunk 1 above)
Comment (danielailie) [reply src/tokenManagement/tokenManagementTransactionsFactory.ts]: good catch :+1:

File: src/tokenManagement/tokenManagementTransactionsFactory.ts (hunk 4 above)
Comment (danielailie) [reply src/tokenManagement/tokenManagementTransactionsFactory.ts]: Updated

File: src/tokenManagement/tokenManagementController.ts (hunk 8 above)
Comment (danielailie) [reply src/tokenManagement/tokenManagementController.ts:4]: I will to this when I will remove the package

File: src/testutils/contractController.ts [hunk 10]
```diff
@@ -15,7 +15,7 @@ export class ContractController {
         this.provider = provider;
         this.transactionCompletionAwaiter = new TransactionWatcher({
             getTransaction: async (hash: string) => {
-                return await provider.getTransaction(hash, true);
+                return await provider.getTransaction(hash);
```
Comment (andreibancioiu) [inline src/testutils/contractController.ts:18]: Can we pass the network provider directly?

File: src/testutils/networkProviders.ts [hunk 11]
```diff
@@ -39,7 +33,7 @@ export function createMainnetProvider(): INetworkProvider {
 export interface INetworkProvider {
     getNetworkConfig(): Promise<INetworkConfig>;
     getAccount(address: IAddress): Promise<IAccountOnNetwork>;
-    getTransaction(txHash: string, withProcessStatus?: boolean): Promise<ITransactionOnNetwork>;
+    getTransaction(txHash: string): Promise<TransactionOnNetwork>;
```
Comment (andreibancioiu) [inline src/testutils/networkProviders.ts:36]: Indeed, good to drop this flag.

File: src/tokenManagement/tokenManagementController.ts (hunk 9 above)
Comment (danielailie) [reply src/tokenManagement/tokenManagementController.ts:477]: done

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**